### ISSUE TITLE
New fake player DOS

### ISSUE THREAD

**Issue by T-Maxxx** [callofduty4x/CoD4x_Server#185]
https://cod4x.me/index.php?/forums/topic/1985-fake-players-spam-protection/&tab=comments#comment-8590

**Comment by T-Maxxx** [comment]
Limit to 2-3 connections per IP? (1 if set allowdupe to dev mode only)
Fast reject for banned IP?

**Comment by icekobrin** [comment]
https://github.com/callofduty4x/CoD4x_Server/pull/159
Им решается проблема. Но ее не ту в 2055 версии сервера.

**Comment by T-Maxxx** [comment]
Это хорошо. Судя по сообщениям, фейковые игроки полностью проходят аутентификацию, что может стать проблемой в будущем.

**Comment by AlexanderCurl** [comment]
It would be useful if this function can be adjusted as a dvar. Define a new dvar: sv_connectionlimit this describes how many clients allowed from the same IP address, and another one: sv_ddosprotection 0-1 enable or disable the function.

**Comment by D4edalus** [comment]
@T-Maxxx the log on that forum post suggests that the fake players are in zombie state. ice kobrins pull request was after the latest server release. i suppose that change just isn't yet distributed.

**Comment by AlexanderCurl** [comment]
@D4edalus these fake clients does even have playerID?

**Comment by D4edalus** [comment]
once fully connected they have to have at least one form of identification. i am not sure about zombie state, but zombie state is not an issue as that is filtered by icekobrins pull request already. however, i suppose that pullrequest would also kick legit players from the same ip after a mapchange ( that last part is just speculation ).

**Comment by AlexanderCurl** [comment]
@D4edalus now how it is works, because I know how it is working, because if me and my wife connecting as a same time, 1 client have been dropped, need to reconnect now. Mapchange not affected because of the loading time.

**Comment by D4edalus** [comment]
the problem you and your wife have doesnt really tell us much, of course one client gets dropped. it always has been like this. what has been added is that more client states trigger the drop. 

tl;dr i believe using the latest server version should be enough to fix volkovs problem

**Comment by AlexanderCurl** [comment]
Latest master branch braking so much stuff, I'm compiled today, my DR killcam gone, lags, etc...

**Comment by D4edalus** [comment]
@AlexanderCurl that's offtopic. but add it to the list: https://github.com/callofduty4x/CoD4x_Server/issues/184

**Comment by D4edalus** [comment]
#159
